lib/std/zig/system.zig

```diff
@@ -221,7 +222,44 @@ pub const NativeTargetInfo = struct {
                     }
                 },
                 .windows => {
-                    // TODO Detect native operating system version.
+                    var version_info: std.os.windows.RTL_OSVERSIONINFOW = undefined;
+                    version_info.dwOSVersionInfoSize = @sizeOf(@TypeOf(version_info));
+
+                    const rc = std.os.windows.ntdll.RtlGetVersion(&version_info);
+                    switch (rc) {
+                        .SUCCESS => {},
+                        else => return std.os.windows.unexpectedStatus(rc),
```

Comment:
Could be:
```zig
                    switch (std.os.windows.ntdll.RtlGetVersion(&version_info)) {
                        .SUCCESS => {},
                        else => |rc| return std.os.windows.unexpectedStatus(rc),
```

Reply:
Oh that's much better, I keep forgetting that `else` is so nice

Reply:
The build failure is linked to this `Unexpected` error being raised here, I don't know if we want to block it before it crosses the stage1/stage2 boundary or make it bubble up all the way to the top.

Reply:
Alternatively you could get this info directly from the [PEB](https://en.wikipedia.org/wiki/Process_Environment_Block) `OSMajorVersion`,`OSMinorVersion`, etc.

Wouldn't have to deal with `unexpected` then. Though I have a hunch the unexpected is unreachable anyway....

Reply:
I guess we have to trust MSDN words `RtlGetVersion returns STATUS_SUCCESS.` :)

Reply:
Sent a PR with the PEB bits you need: https://github.com/ziglang/zig/pull/4586/files#diff-1dd663aea8f27c93da28521ba3f342d2R1242-R1244

Reply:
FYI if you ever get an `Unexpected` error then a new switch case needs to be added where it says `return unexpectedError()`. Maybe it should print something like this to indicate

Reply:
Also if this information is in the PEB, that's my vote for where to get the info. So far we haven't had any issues with the PEB.

Reply:
This API cannot fail and requires no allocator, I don't really like pulling stuff from the undocumented *EB structures as those are not part of the ABI/API. Windows has to cope with a lot of backward compatibility shenanigans because people love{d,} to pull this kind of tricks, I don't want to be part of the problem.